[package] cpython/*: CMake intergration break with conan 2.21.0

### Description

The find package I use
```cmake
find_package(Python3 REQUIRED CONFIG OPTIONAL_COMPONENTS Interpreter Development)
```

fail under conan 2.21.0 while it was working before that.

it might be caused by a change to the new CMake deps generator as I use:
`tools.cmake.cmakedeps:new=will_break_next`


### Package and Environment Details

* Package Name/Version: **cpython/3.10.18**
* Operating System+version: **Windows and Rocky 8**
* Compiler+version: **MSVC193 / Icpc **
* Python version: **Python 3.10.12**


### Conan profile

```
{% set prefix, arch, compiler, compiler_ver, compiler_update, winsdk_ver = profile_name.split('-') %}
[buildenv]
#python ignores tools.microsoft:winsdk_version but listen to this variable
DefaultWindowsSDKVersion={{winsdk_ver}}

[settings]
arch={{ arch }}
build_type=Release
compiler={{compiler}}
compiler.cppstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.update={{compiler_update}}
compiler.version={{compiler_ver}}
os=Windows

cpython/*: build_type=Release
cpython/*: compiler.runtime_type=Release

[options]
openssl/*:no_module=True
openssl/*:no_zlib=True

cpython/*:with_gdbm=False
cpython/*:with_nis=False
cpython/*:with_lzma=False
cpython/*:with_tkinter=False
cpython/*:with_curses=False
cpython/*:shared=True

[conf]
tools.microsoft:winsdk_version={{winsdk_ver}}
tools.cmake.cmakedeps:new=will_break_next
```

### Steps to reproduce

Install the cpython conan package with the new cmake deps generator.

Run the following cmake(i use cmake/3.31.6) code

```
find_package(Python3 REQUIRED CONFIG OPTIONAL_COMPONENTS Interpreter Development)
```

it should fail

### Logs

<details><summary>cmake output with conan 2.21.0</summary>

```
-- Conan: Target declared imported INTERFACE library 'cpython::python'
[cmake] -- Conan: Target declared imported SHARED library 'cpython::embed'
[cmake] -- Conan: Target declared imported INTERFACE library 'cpython::cpython'
[cmake] -- Conan: Including build module from 'C:/Users/mbarbier/.conan2/p/cpyth27ca53cabb780/p/bin/libs/cmake/use_conan_python.cmake'
[cmake] CMake Warning (dev) at C:/Users/mbarbier/.conan2/p/cmakead94e7ce16794/p/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:441 (message):
[cmake]   The package name passed to `find_package_handle_standard_args` (Python)
[cmake]   does not match the name of the calling package (Python3).  This can lead to
[cmake]   problems in calling code that expects `find_package` result variables
[cmake]   (e.g., `_FOUND`) to follow a certain pattern.
[cmake] Call Stack (most recent call first):
[cmake]   C:/Users/mbarbier/.conan2/p/cmakead94e7ce16794/p/share/cmake-3.31/Modules/FindPython/Support.cmake:4002 (find_package_handle_standard_args)
[cmake]   C:/Users/mbarbier/.conan2/p/cmakead94e7ce16794/p/share/cmake-3.31/Modules/FindPython.cmake:652 (include)
[cmake]   C:/Users/mbarbier/.conan2/p/cpyth27ca53cabb780/p/bin/libs/cmake/use_conan_python.cmake:14 (include)
[cmake]   conan/Release/Python3Config.cmake:16 (include)
[cmake]   _SpeosCMake/PythonVenv.cmake:7 (find_package)
[cmake]   _SpeosCMake/SpeosCMake.cmake:44 (include)
[cmake]   CMakeLists.txt:21 (speoscmake_use_python)
[cmake] This warning is for project developers.  Use -Wno-dev to suppress it.
[cmake] 
[cmake] -- Configuring incomplete, errors occurred!
[cmake] CMake Warning (dev) at C:/Users/mbarbier/.conan2/p/cmakead94e7ce16794/p/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:441 (message):
[cmake]   The package name passed to `find_package_handle_standard_args` (Python)
[cmake]   does not match the name of the calling package (Python3).  This can lead to
[cmake]   problems in calling code that expects `find_package` result variables
[cmake]   (e.g., `_FOUND`) to follow a certain pattern.
[cmake] Call Stack (most recent call first):
[cmake]   C:/Users/mbarbier/.conan2/p/cmakead94e7ce16794/p/share/cmake-3.31/Modules/FindPython.cmake:673 (find_package_handle_standard_args)
[cmake]   C:/Users/mbarbier/.conan2/p/cpyth27ca53cabb780/p/bin/libs/cmake/use_conan_python.cmake:14 (include)
[cmake]   conan/Release/Python3Config.cmake:16 (include)
[cmake]   _SpeosCMake/PythonVenv.cmake:7 (find_package)
[cmake]   _SpeosCMake/SpeosCMake.cmake:44 (include)
[cmake]   CMakeLists.txt:21 (speoscmake_use_python)
[cmake] This warning is for project developers.  Use -Wno-dev to suppress it.
[cmake] 
[cmake] CMake Error at C:/Users/mbarbier/.conan2/p/cpyth27ca53cabb780/p/bin/libs/cmake/use_conan_python.cmake:18 (message):
[cmake]   CMake detected wrong cpython version - this is likely a bug with the
[cmake]   cpython Conan package
[cmake] Call Stack (most recent call first):
[cmake]   conan/Release/Python3Config.cmake:16 (include)
[cmake]   _SpeosCMake/PythonVenv.cmake:7 (find_package)
[cmake]   _SpeosCMake/SpeosCMake.cmake:44 (include)
[cmake]   CMakeLists.txt:21 (speoscmake_use_python)
[cmake] 
[cmake] 
```

</details>

<details><summary>cmake output with conan 2.20.1</summary>

```
[cmake] -- Conan: Configuring Targets for cpython/3.10.18
[cmake] -- Conan: Configuring Targets for openssl/3.5.2
[cmake] -- Conan: Target declared imported STATIC library 'OpenSSL::SSL'
[cmake] -- Conan: Target declared imported STATIC library 'OpenSSL::Crypto'
[cmake] -- Conan: Target declared imported INTERFACE library 'openssl::openssl'
[cmake] -- Conan: Including build module from 'C:/Users/mbarbier/.conan2/p/opens297955996cc49/p/lib/cmake/conan-official-openssl-variables.cmake'
[cmake] -- Conan: Configuring Targets for zlib/1.3.1
[cmake] -- Conan: Target declared imported STATIC library 'ZLIB::ZLIB'
[cmake] -- Conan: Target declared imported INTERFACE library 'cpython::python'
[cmake] -- Conan: Target declared imported SHARED library 'cpython::embed'
[cmake] -- Conan: Target declared imported INTERFACE library 'cpython::cpython'
[cmake] -- Conan: Including build module from 'C:/Users/mbarbier/.conan2/p/cpyth27ca53cabb780/p/bin/libs/cmake/use_conan_python.cmake'
[cmake] -- Found Python3:  (found version "3.10.18") found components: Interpreter Development Development.Module Development.Embed
```

</details>

--- a/conan/tools/cmake/cmakedeps2/config.py
+++ b/conan/tools/cmake/cmakedeps2/config.py
@@ -124,12 +124,6 @@ class ConfigTemplate2:
                                "adding the '-DCMAKE_BUILD_TYPE=<build_type>' argument.")
         endif()
 
-        # build_modules_paths comes from last configuration only
-        {% for build_module in build_modules_paths %}
-        message(STATUS "Conan: Including build module from '{{build_module}}'")
-        include("{{ build_module }}")
-        {% endfor %}
-
         {% if components %}
         set({{filename}}_PACKAGE_PROVIDED_COMPONENTS {{components}})
         foreach(comp {%raw%}${{%endraw%}{{filename}}_FIND_COMPONENTS})
@@ -155,6 +149,14 @@ class ConfigTemplate2:
         {% if definitions is not none %}
         set({{ prefix }}_DEFINITIONS {{ definitions}} )
         {% endif %}
+        {% endfor %}
+
+        # build_modules_paths comes from last configuration only
+        # Some build modules in ConanCenter use try_compile variables and legacy, so this
+        # include() needs to happen after the above variables are defined
+        {% for build_module in build_modules_paths %}
+        message(STATUS "Conan: Including build module from '{{build_module}}'")
+        include("{{ build_module }}")
         {% endfor %}
 
         # Definition of extra CMake variables from cmake_extra_variables

